server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/FetchHandler.scala

```diff
@@ -0,0 +1,164 @@
+            s" $startMapIndex endMapIndex $endMapIndex is empty.")
+        }
+        client.getChannel.writeAndFlush(new RpcResponse(request.requestId,
+          new NioManagedBuffer(streamHandle.toByteBuffer)))
+      } catch {
+        case e: IOException =>
+          client.getChannel.writeAndFlush(new RpcFailure(request.requestId,
+            Throwables.getStackTraceAsString(new RssException("Chunk offsets meta exception ", e))))
+      } finally {
+        // metrics end
+        source.stopTimer(WorkerSource.OpenStreamTime, shuffleKey)
+        request.body().release()
+      }
+    } else {
+      source.stopTimer(WorkerSource.OpenStreamTime, shuffleKey)
+      client.getChannel.writeAndFlush(new RpcFailure(request.requestId,
+        Throwables.getStackTraceAsString(new FileNotFoundException)))
+    }
+  }
+
+  def handleChunkFetchRequest(client: TransportClient, req: ChunkFetchRequest): Unit = {
+    source.startTimer(NetWorkSource.FetchChunkTime, req.toString)
+    logTrace(s"Received req from ${NettyUtils.getRemoteAddress(client.getChannel)}" +
+      s" to fetch block ${req.streamChunkSlice}")
+
+    val chunksBeingTransferred = streamManager.chunksBeingTransferred
+    if (chunksBeingTransferred >= conf.maxChunksBeingTransferred) {
+      logError(s"The number of chunks being transferred $chunksBeingTransferred" +
+        s"is above ${conf.maxChunksBeingTransferred()}.")
+      source.stopTimer(NetWorkSource.FetchChunkTime, req.toString)
+    } else {
+      try {
+        streamManager.checkAuthorization(client, req.streamChunkSlice.streamId)
+
+        val buf = streamManager.getChunk(req.streamChunkSlice.streamId,
+          req.streamChunkSlice.chunkIndex, req.streamChunkSlice.offset, req.streamChunkSlice.len)
+        streamManager.chunkBeingSent(req.streamChunkSlice.streamId)
+        client.getChannel.writeAndFlush(new ChunkFetchSuccess(req.streamChunkSlice, buf))
+          .addListener(new GenericFutureListener[Future[_ >: Void]] {
+            override def operationComplete(future: Future[_ >: Void]): Unit = {
+              streamManager.chunkSent(req.streamChunkSlice.streamId)
+              source.stopTimer(NetWorkSource.FetchChunkTime, req.toString)
+            }
+          })
+      } catch {
+        case e: Exception =>
+          logError(String.format(s"Error opening block ${req.streamChunkSlice} for request from" +
+            s" ${NettyUtils.getRemoteAddress(client.getChannel)}", e))
+          client.getChannel.writeAndFlush(new ChunkFetchFailure(req.streamChunkSlice,
+            Throwables.getStackTraceAsString(e)))
+          source.stopTimer(NetWorkSource.FetchChunkTime, req.toString)
+      }
+    }
+  }
+
+  override def checkRegistered: Boolean = registered.get
+
+  override def channelInactive(client: TransportClient): Unit = {
+    logDebug("channel Inactive " + client.getSocketAddress)
+  }
+
+  override def exceptionCaught(cause: Throwable, client: TransportClient): Unit = {
```

Comment:
The same with above.

Reply:
I think it's fine to reserve these methods.